src/stumpy_png/png.cr

```diff
@@ -67,43 +57,46 @@ module StumpyPNG
       end
     end
 
-    def parse_IDAT(chunk : Chunk)
+    def parse_idat(chunk : Chunk)
       @idat_count += 1
       @idat_buffer.write(chunk.data)
     end
 
-    def parse_PLTE(chunk : Chunk)
+    def parse_plte(chunk : Chunk)
       raise "Invalid palette length" unless (chunk.size % 3) == 0
       @palette = chunk.data.each_slice(3).map { |rgb| RGBA.from_rgb_n(rgb, 8) }.to_a
     end
 
-    def parse_IHDR(chunk : Chunk)
+    def parse_ihdr(chunk : Chunk)
       @width = Utils.parse_integer(chunk.data[0, 4])
       @height = Utils.parse_integer(chunk.data[4, 4])
 
-      @bit_depth = chunk.data[8]
       @color_type = chunk.data[9]
-      raise "Invalid color type" unless COLOR_TYPES.has_key?(@color_type)
-      unless COLOR_TYPES[@color_type][1].includes?(@bit_depth)
+      color_type = COLOR_TYPES[@color_type]?
```

Comment:
I've not kept up with crystal programming style,
is `||` considered the idiomatic way to do this?

Reply:
It's one of many I guess, just bit more succinct version if `if ... end` construct.